pybossa/view/account.py

```diff
@@ -500,6 +503,16 @@ def create_account(user_data, project_slugs=None, ldap_disabled=True, auto_creat
     msg = generate_invitation_email_for_new_user(user=user_info, project_slugs=project_slugs)
     mail_queue.enqueue(send_mail, msg)
 
+    if auto_create and user_data.get("data_access_type", None) == "external":
+        # if the account is created automatically and has external data access, send warning
+        admins_email_list = [u.email_addr for u in user_repo.filter_by(admin=True)]
+        admin_msg = generate_bsso_account_notification(user=user_data, admins_emails=admins_email_list, access_type="BSSO", warning=True)
+        mail_queue.enqueue(send_mail, admin_msg)
+    elif auto_create:
+        # if the account is just automatically created, just send notification
+        alert_msg = generate_bsso_account_notification(user=user_info, admins_emails=current_app.config.get('ALERT_LIST',[]), access_type="BSSO")
+        mail_queue.enqueue(send_mail, alert_msg) 
```

Comment:
overall, i think there's scope of reusing existing variables such as from pybossa.data_access import data_access_levels, user_info['user_type'] that would simplify this logic. I'll get in touch with you and submit gist w/ sample refactored code for reference. Thanks!

Reply:
https://gist.github.com/dchhabda/82ec93287d4c303ffbdd9061863edb63